### PR TITLE
Delay transition

### PR DESCRIPTION
**Pull request by darrenburns**
Stacked on top of #683 - review that one first.

### CODE REVIEW

**Review by willmcgugan — CHANGES_REQUESTED** [review]
This solution will invoke the callback during the "delay" period. The end result will be correct, in that the user will see the delay. The downside is that the animator is working during the delay period and not updating anything.

Ideally the animator would not do any work until the delay has expired.

If this looks particularly complex to implement, suggest putting in a early-out condition. i.e. if the delay hasn't expired return as early as possible.

File: src/textual/_animator.py [hunk 1]
```diff
@@ -141,7 +143,7 @@ def __call__(
 class Animator:
     """An object to manage updates to a given attribute over a period of time."""
 
-    def __init__(self, target: MessageTarget, frames_per_second: int = 60) -> None:
+    def __init__(self, target: App, frames_per_second: int = 60) -> None:
```
Comment (willmcgugan) [inline src/textual/_animator.py:146]: Is the Animator only associated with the app?

If so, may as well rename `target` to `app` ?

**Review by willmcgugan — APPROVED** [review]
Just one thing to consider.

File: src/textual/_animator.py (hunk 1 above)
Comment (darrenburns) [reply src/textual/_animator.py]: Done

**Pull request merged**